docs/guide/api-javascript.md

```diff
@@ -87,16 +87,6 @@ interface ViteDevServer {
    * Apply Vite built-in HTML transforms and any plugin HTML transforms.
    */
   transformIndexHtml(url: string, html: string): Promise<string>
-  /**
-   * Util for transforming a file with esbuild.
-   * Can be useful for certain plugins.
-   */
-  transformWithEsbuild(
-    code: string,
-    filename: string,
-    options?: EsbuildTransformOptions,
-    inMap?: object
-  ): Promise<ESBuildTransformResult>
```

Comment:
I think we should add the description of the new free `transformWithEsbuild` function to the docs

Reply:
I can add it, but there's a lot of other APIs not documented currently in the [Javascript API docs](https://vitejs.dev/guide/api-javascript.html) too. So I thought `transformWithEsbuild` should follow suit.

Reply:
Since plugin-svelte would use this one and it was documented before, I think it is better in this case to keep a minimal reference to it in the docs

Reply:
@patak-js I've added the docs. Along the way I also found that we need to export the `EsbuildTransformOptions` type, so I did that as well, otherwise that type would be `TransformOptions_2`